src/command_helpers.rs

```diff
@@ -311,6 +311,14 @@ pub(crate) fn objects_from_files(files: &[Arc<Path>], dst: &Path) -> Result<Vec<
         // Hash the dirname. This should prevent conflicts if we have multiple
         // object files with the same filename in different subfolders.
         let mut hasher = hash_map::DefaultHasher::new();
+
+        // Make the dirname relative to avoid full system paths influencing the sha and
```

Comment:
I'm ok with making it relative to cargo workspace root

Alternatively maybe hashing the entire file is also a good idea, but:
 - it would then be opt-in (because extra dep)
 - attackers can change the file while we are calculating hashsum/compiling

Reply:
> hashing the entire file is also a good idea

I like that actually. Up to you!

Reply:
> I like that actually. Up to you!

In that case, I would like you to add an optional feature `sha1-file-hashing` which enables dep [`sha1-checked`](https://lib.rs/crates/sha1-checked).

Reply:
@NobodyXu actually upon further reflection, I think it might be nice if things work out of the box, i.e. without needed to enable a feature. I've updated the change to make it relative to the workspace root:

> I'm ok with making it relative to cargo workspace root

PTAL!